File: src/HotelOrderSystem.Api/Dtos/AuthDtos.cs
using System.ComponentModel.DataAnnotations;

namespace HotelOrderSystem.Api.Dtos;

public sealed record LoginRequest(
    [property: Required, MaxLength(80)] string UserName,
    [property: Required, MaxLength(200)] string Password);

public sealed record AuthResponse(string Token, DateTime ExpiresAtUtc, UserProfileDto User);

public sealed record UserProfileDto(
    int UserId,
    string FullName,
    string UserName,
    string Role,
    int? TeamId,
    string? TeamName);

public sealed record DeviceTokenRequest(
    [property: Required, MaxLength(200)] string DeviceId,
    [property: Required, MaxLength(50)] string Platform,
    [property: MaxLength(50)] string AppVersion,
    [property: MaxLength(1000)] string FcmToken);

public sealed record LogoutRequest(
    [property: Required, MaxLength(200)] string DeviceId);

File: android-staff-app/app/src/main/java/com/ibaapps/HotelOrderSystem/data/remote/dto/AuthDtos.kt
package com.ibaapps.HotelOrderSystem.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class LoginRequestDto(
    val userName: String,
    val password: String
)

@Serializable
data class AuthResponseDto(
    val token: String,
    val expiresAtUtc: String,
    val user: UserProfileDto
)

@Serializable
data class UserProfileDto(
    val userId: Int,
    val fullName: String,
    val userName: String,
    val role: String,
    val teamId: Int? = null,
    val teamName: String? = null
)

@Serializable
data class DeviceTokenRequestDto(
    val deviceId: String,
    val platform: String,
    val appVersion: String,
    val fcmToken: String
)

@Serializable
data class LogoutRequestDto(
    val deviceId: String
)
